src/UI/PickLevel.hs

```diff
@@ -11,17 +12,39 @@ import qualified Brick.Widgets.Border.Style as BS
 import qualified Brick.Widgets.Center as C
 import qualified Graphics.Vty as V
 
-app :: App (Maybe Int) e ()
+data LevelConfig = LevelConfig
+  { levelNumber :: Int
+  , progression :: Bool
+  } deriving (Show, Eq)
+
+data MenuOption = YesOption | NoOption deriving (Eq)
+
+data PickState = PickState
+  { currentLevel :: Maybe Int
+  , showProgression :: Bool
+  , pickingLevel :: Bool
+  , selectedOption :: MenuOption
+  }
+
+app :: App PickState e ()
 app = App
-  { appDraw         = const [ui]
+  { appDraw         = drawUI
   , appHandleEvent  = handleEvent
   , appStartEvent   = pure ()
-  , appAttrMap      = const $ attrMap V.defAttr []
+  , appAttrMap      = const $ attrMap V.defAttr
+      [ (selectedAttr, V.black `on` V.white)
+      ]
   , appChooseCursor = neverShowCursor
   }
 
-ui :: Widget ()
-ui =
+selectedAttr :: AttrName
+selectedAttr = attrName "selected"
+
+drawUI :: PickState -> [Widget ()]
+drawUI ps = [ui ps]
```

Comment:
I just tried this out and there are a few funky bits:

![Screenshot 2024-12-06 at 10 43 51 AM](https://github.com/user-attachments/assets/edd24afe-e662-49da-a033-54ddc9417326)
- Menu not centered
- doesn't respect j/k bindings
- and the Enter prompt is cutoff

My recommendation: drop the changes to this level picker UI (sorry) and only expose this option in the CLI. See the `high-score` option for an example of a boolean flag.

Reply:
I do agree with the other changes, but I kinda wanna disagree here;
- For anyone who isn't familiar with the docs, CLI args are a relative unknown, and both game modes are equally fun ways of playing. I believe that this should definitely be an option reflected in the GUI/TUI, as that's what separates your Tetris apart. It's intuitive and usable by a newb.
- Fixed centering and j/k binding.

Reply:
CLI arguments are generally very common and discoverable in the world of software development, but there might be an audience for tetris that is not quite as technical. So, I'm cool with keeping this as you suggest 👍 After I merge I'll add a CLI arg that will skip this prompt if set.

Reply:
Oh, I absolutely agree. I use args quite a bit (am a homelab sysadmin); I meant that the specific args for this game need to be read up by a user, and that's unlikely. 

 thank you very much!